Arduino Uno R3/libraries/ADE9153A/ADE9153AAPI.cpp

```diff
@@ -0,0 +1,435 @@
+*/
+
+void ADE9153AClass:: ReadEnergyRegs(EnergyRegs *Data)
+{
+	int32_t tempReg;
+	float tempValue;
+	
+	tempReg = int32_t (SPI_Read_32(REG_AWATTHR_HI));
+	Data->ActiveEnergyReg = tempReg;
+	tempValue = (float)tempReg * CAL_ENERGY_CC / 1000;
+	Data->ActiveEnergyValue = tempValue;				//Energy in mWhr
+	
+	tempReg = int32_t (SPI_Read_32(REG_AFVARHR_HI));
+	Data->FundReactiveEnergyReg = tempReg;
+	tempValue = (float)tempReg * CAL_ENERGY_CC / 1000;
+	Data->FundReactiveEnergyValue = tempValue;			//Energy in mVARhr
+	
+	tempReg = int32_t (SPI_Read_32(REG_AVAHR_HI));
+	Data->ApparentEnergyReg = tempReg;
+	tempValue = (float)tempReg * CAL_ENERGY_CC / 1000;
+	Data->ApparentEnergyValue = tempValue;				//Energy in mVAhr
+}
+
+void ADE9153AClass:: ReadPowerRegs(PowerRegs *Data)
+{
+	int32_t tempReg;
+	float tempValue;
+	
+	tempReg = int32_t (SPI_Read_32(REG_AWATT));
+	Data->ActivePowerReg = tempReg;
+	tempValue = (float)tempReg * CAL_POWER_CC / 1000;
+	Data->ActivePowerValue = tempValue;					//Power in mW
+	
+	tempReg = int32_t (SPI_Read_32(REG_AFVAR));
+	Data->FundReactivePowerReg = tempReg;
+	tempValue = (float)tempReg * CAL_POWER_CC / 1000;
+	Data->FundReactivePowerValue = tempValue;			//Power in mVAR
+	
+	tempReg = int32_t (SPI_Read_32(REG_AVA));
+	Data->ApparentPowerReg = tempReg;
+	tempValue = (float)tempReg * CAL_POWER_CC / 1000;
+	Data->ApparentPowerValue = tempValue;				//Power in mVA
+}
+
+void ADE9153AClass:: ReadRMSRegs(RMSRegs *Data)
+{
+	uint32_t tempReg;
+	float tempValue;
+	
+	tempReg = int32_t (SPI_Read_32(REG_AIRMS));
+	Data->CurrentRMSReg = tempReg;
+	tempValue = (float)tempReg * CAL_IRMS_CC / 1000;	//RMS in mA
+	Data->CurrentRMSValue = tempValue;
+	
+	tempReg = int32_t (SPI_Read_32(REG_AVRMS));
+	Data->VoltageRMSReg = tempReg;
+	tempValue = (float)tempReg * CAL_VRMS_CC / 1000;	//RMS in mV
+	Data->VoltageRMSValue = tempValue;
+}
+
+void ADE9153AClass:: ReadHalfRMSRegs(HalfRMSRegs *Data)
+{
+	uint32_t tempReg;
+	float tempValue;
+	
+	tempReg = int32_t (SPI_Read_32(REG_AIRMS_OC));
+	Data->HalfCurrentRMSReg = tempReg;
+	tempValue = (float)tempReg * CAL_IRMS_CC / 1000;	//Half-RMS in mA
+	Data->HalfCurrentRMSValue = tempValue;
+	
+	tempReg = int32_t (SPI_Read_32(REG_AVRMS_OC));
+	Data->HalfVoltageRMSReg = tempReg;
+	tempValue = (float)tempReg * CAL_VRMS_CC / 1000;	//Half-RMS in mV
+	Data->HalfVoltageRMSValue = tempValue;
+}
+
+void ADE9153AClass:: ReadPQRegs(PQRegs *Data)
+{
+	int32_t tempReg;
+	uint16_t temp;
+	float mulConstant;
+	float tempValue;
+	
+	tempReg=int32_t (SPI_Read_32(REG_APF)); //Read PF register
+	Data->PowerFactorReg = tempReg;
+	tempValue=(float)tempReg/(float)134217728; //Calculate PF
+	Data->PowerFactorValue=tempValue;
+	
+	tempReg=int32_t (SPI_Read_32(REG_APERIOD)); //Read PERIOD register
+	Data->PeriodReg = tempReg;
+	tempValue=(float)(4000*65536)/(float)(tempReg+1); //Calculate Frequency
+	Data->FrequencyValue = tempValue;
+	
+	temp=SPI_Read_16(REG_ACCMODE); //Read frequency setting register
+	if((temp&0x0010)>0)

```

Comment:
we use linux coding style so the if will be like this:
if () {
     foo();
} else {
     bar();
}

Reply:
add some spaces in if ((temp & 0x0010) > 0)

Reply:
also check for this in other places also a * b. It is easier to read code like this.